network/transport/grpc/connection_manager.go

```diff
@@ -458,10 +592,17 @@ func (s *grpcConnectionManager) handleInboundStream(protocol Protocol, inboundSt
 	return nil
 }
 
-func (s *grpcConnectionManager) constructMetadata() (metadata.MD, error) {
-	md := metadata.New(map[string]string{
-		peerIDHeader: string(s.config.peerID),
-	})
+func (s *grpcConnectionManager) constructMetadata(bootstrap bool) (metadata.MD, error) {
+	md := metadata.New(map[string]string{})
+
+	if bootstrap {
+		// Older nodes (< v5.1) only match on peerID.
+		// The postfix allows them to have a bootstrap connection and authenticated connection at the same time.
+		md.Set(peerIDHeader, string(s.config.peerID)+"-bootstrap")
```

Comment:
so we're postfixing our own ID with `-bootstrap`, wouldn't it make more sense to postfix the peer's ID as we receive it, from a bootstrap node? (meaning in handleInbound)?

Reply:
This is mainly to allow a bootstrap and an authenticated connection at the same time for older nodes that deduplicate on peerID. Without the postfix the other side would drop one of the two.

handeling inbound we do not know if the peer is setting up a bootstrap connection to us or just doesn't have a DID (yet)

Reply:
I will postfix the received peerID as well when creating a bootstrap stream. This should result in 1 less bootstrap stream if both nodes have configured each other as bootstrap node.